interview-questions copy.md

```diff
@@ -0,0 +1,74 @@
+# ASSESSMENT 6: Interview Practice Questions
+Answer the following questions.
+
+First, without external resources. Challenge yourself to answer from memory.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
+
+  Your answer: The foreign key in this example would be located in the Sightings model.  The foreign key is animal_id, because it is an attribute that refers to to the specific animal in our model Animal that it belongs to.  When dealing with foreign keys, we know that these are attributes that references a candidate key
+
+  Researched answer: A perfect example for this explanation is if you have a table for a Team would likely have member_name in it.  That member_name would likely have many People in it.  People need names, and those People in the People model are team members.  So the key is member_name because it is likely referring  to the person_name key in the model.
+
+
+
+2. Which RESTful routes must always be passed params? Why?
+
+  Your answer: RESTful routes that must always be passed params are GET and DELETE.  In order for either HTTP verb to fully function, we must pass params into it in order to specifically target the method we wish to target in order to complete the action we wish to complete.
+
+  Researched answer: Specifically, params refers to the parameters being passed to the controller via a GET or POST request.
+
```

Comment:
GET and DELETE are http verbs. While technically they have routes that require params (delete must be passed an id and show, which falls under the GET action also requires an id) the verbs are not *technically* the same thing as the routes. Routes are what gets appended to the url to specify the request.